x-pack/plugins/security_solution/scripts/endpoint/action_responder/run_in_auto_mode.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { RunContext } from '@kbn/dev-cli-runner';
+import { set } from 'lodash';
+import { SUPPORTED_TOKENS } from './constants';
+import { ActionDetails } from '../../../common/endpoint/types';
+import { createRuntimeServices, RuntimeServices } from '../common/stack_services';
+
+import {
+  fetchEndpointActionList,
+  sendEndpointActionResponse,
+  sendFleetActionResponse,
+  sleep,
+} from './utils';
+
+const ACTION_RESPONSE_DELAY = 40_000;
+
+export const runInAutoMode = async ({
+  log,
+  flags: { username, password, asSuperuser, kibana, elastic, delay: _delay },
+}: RunContext) => {
+  const runtimeServices = await createRuntimeServices({
+    log,
+    password: password as string,
+    username: username as string,
+    asSuperuser: asSuperuser as boolean,
+    elasticsearchUrl: elastic as string,
+    kibanaUrl: kibana as string,
+  });
+
+  log.write(`  ${SUPPORTED_TOKENS}`);
+
+  const delay = Number(_delay) || ACTION_RESPONSE_DELAY;
+
+  do {
+    await checkPendingActionsAndRespond(runtimeServices, { delay });
+    await sleep(5_000);
+  } while (true);
+};
+
+const checkPendingActionsAndRespond = async (
+  { kbnClient, esClient, log }: RuntimeServices,
+  { delay = ACTION_RESPONSE_DELAY }: { delay?: number } = {}
```

Comment:
Should this default value be `{delay = ACTION_RESPONSE_DELAY}`?

Reply:
yeah, the default is 40s. it can be overwritten via CLI input `--delay`. Or did I miss-understand you? 🤔

Reply:
It was my fault, that's ok 🙂